it/google-cloud-platform/src/test/java/com/google/cloud/teleport/it/gcp/bigtable/BigTableIOLT.java

```diff
@@ -0,0 +1,251 @@
+    assertEquals(configuration.getNumRows(), numRecords, 0.5);
+  }
+
+  private PipelineLauncher.LaunchInfo testWrite() throws IOException {
+
+    BigtableIO.Write writeIO =
+        BigtableIO.write().withBigtableOptions(bigtableOptions).withTableId(tableId);
+
+    writePipeline
+        .apply(GenerateSequence.from(0).to(configuration.getNumRows()))
+        .apply(ParDo.of(new MapToBigTableFormat()))
+        .apply("Write to BigTable", writeIO);
+
+    PipelineLauncher.LaunchConfig options =
+        PipelineLauncher.LaunchConfig.builder("test-bigtable-write")
+            .setSdk(PipelineLauncher.Sdk.JAVA)
+            .setPipeline(writePipeline)
+            .addParameter("runner", configuration.getRunner())
+            .build();
+
+    return pipelineLauncher.launch(project, region, options);
+  }
+
+  private PipelineLauncher.LaunchInfo testRead() throws IOException {
+    BigtableIO.Read readIO =
+        BigtableIO.read()
+            .withoutValidation()
+            .withProjectId(project)
+            .withInstanceId(resourceManager.getInstanceId())
+            .withTableId(tableId);
+
+    readPipeline
+        .apply("Read from BigTable", readIO)
+        .apply("Counting element", ParDo.of(new CountingFn<>(READ_ELEMENT_METRIC_NAME)));
+
+    PipelineLauncher.LaunchConfig options =
+        PipelineLauncher.LaunchConfig.builder("test-bigtable-read")
+            .setSdk(PipelineLauncher.Sdk.JAVA)
+            .setPipeline(readPipeline)
+            .addParameter("runner", configuration.getRunner())
+            .build();
+
+    return pipelineLauncher.launch(project, region, options);
+  }
+
+  /** Options for Bigquery IO load test. */
+  @AutoValue
+  abstract static class Configuration {
+    abstract Long getNumRows();
+
+    abstract Integer getPipelineTimeout();
+
+    abstract String getRunner();
+
+    static Configuration of(long numRows, int pipelineTimeout, String runner) {
+      return new AutoValue_BigTableIOLT_Configuration.Builder()
+          .setNumRows(numRows)
+          .setPipelineTimeout(pipelineTimeout)
+          .setRunner(runner)
+          .build();
+    }
+
+    @AutoValue.Builder
+    abstract static class Builder {
+      abstract Configuration.Builder setNumRows(long numRows);
+
+      abstract Configuration.Builder setPipelineTimeout(int timeOutMinutes);
+
+      abstract Configuration.Builder setRunner(String runner);
+
+      abstract Configuration build();
+    }
+
+    abstract Configuration.Builder toBuilder();
+  }
+
+  /** Maps long number to the BigTable format record. */
+  private static class MapToBigTableFormat extends DoFn<Long, KV<ByteString, Iterable<Mutation>>>
+      implements Serializable {
+
+    @ProcessElement
+    public void processElement(ProcessContext c) {
+      Long index = c.element();
+
+      ByteString key = ByteString.copyFromUtf8(String.format("key%09d", index));
+      ByteString value = ByteString.copyFromUtf8(String.format("value%09d", index));
```

Comment:
We still need a rowSize in Configuration and assign the value of corresponding size here, this is because we are interested in the MB/s throughput also

For the large preset, please use 100M record and 10GB data size. We would like to compare the performance cross IOs using same size of data also

Reply:
Done,
Used values from BigQueryIOLT